network/wifi/at_parser.c

```diff
@@ -364,47 +375,47 @@ static bool is_async_messages(struct at_desc *desc, uint8_t ch)
 }
 
 /* Update the buffer descriptor when a read transaction ended */
-static inline void end_conn_read(volatile struct at_circular_buff *cbuff)
+static inline void end_conn_read(struct at_desc *desc)
 {
-	cbuff->wi += cbuff->last_len;
-	if (cbuff->wi == cbuff->size)
-		cbuff->wi = 0; //Move wi at the begining of buff
-	if (cbuff->count + cbuff->last_len > cbuff->size) {
-		cbuff->is_overflow = true; //Set overflow
-		cbuff->count = cbuff->size;
-		cbuff->ri = cbuff->wi;
-	} else {
-		cbuff->count += cbuff->last_len;
-	}
-	cbuff->new_data = true;
+	struct connection_desc	*conn;
+
+	conn = &desc->conn[desc->current_conn];
+
+	cb_end_async_write(conn->cbuff);
 }
 
 /* Start new read operation. If the read buffer is nul, aux_buff will be used */
-static inline void start_conn_read(struct connection_desc *conn,
-				   struct uart_desc *uart_desc,
-				   uint8_t *aux_buff)
+static inline void start_conn_read(struct at_desc *desc)
 {
-	uint32_t				len;
-	volatile struct at_circular_buff	*cbuff = &(conn->cbuff);
-
-	if (!cbuff->buff || !cbuff->size) {
-		/* Read new data in aux_buff */
-		uart_read_nonblocking(uart_desc, aux_buff, 1);
-		conn->to_read -= 1;
-		return ;
-	}
+	struct connection_desc	*conn;
+	uint8_t			*buff;
+	uint32_t		available_len;
+	uint32_t		ret;
+
+	conn = &desc->conn[desc->current_conn];
+
+	if (!conn->cbuff)
+		goto dummy_read;
 
-	len = min(cbuff->size - cbuff->wi, conn->to_read);
-	uart_read_nonblocking(uart_desc, cbuff->buff + cbuff->wi, len);
-	conn->to_read -= len;
-	cbuff->last_len = len;
+	ret = cb_prepare_async_write(conn->cbuff, conn->to_read,
+				     (void **)&buff, &available_len);
+	if (IS_ERR_VALUE(ret))
+		goto dummy_read;
+
+	uart_read_nonblocking(desc->uart_desc, buff, available_len);
+	conn->to_read -= available_len;
+
+	return ;
+dummy_read:
+	/* Read new data in aux_buff */
+	uart_read_nonblocking(desc->uart_desc, &desc->read_ch, 1);
```

Comment:
Why is this dummy read required ? Maybe a comment documenting this should be added.

Reply:
In order to consume the data if no buffer available.
Will document it